client/src/partials/vouchers/complex.html

```diff
@@ -0,0 +1,219 @@
+                  <p><strong class="control-label">{{ "FORM.LABELS.CURRENCY" | translate }}</strong></p>
+                  <label ng-repeat="currency in ComplexVoucherCtrl.currencies track by currency.id"
+                    class="radio-inline">
+                    <input
+                      name="currency"
+                      type="radio"
+                      ng-model="ComplexVoucherCtrl.voucher.currency_id"
+                      ng-value="currency.id"
+                      data-currency-option="{{ currency.id }}"
+                      required>
+                    {{ currency.label }}
+                  </label>
+
+                  <div class="help-block" ng-messages="ComplexVoucherForm.currency.$error" ng-show="ComplexVoucherForm.$submitted">
+                    <div ng-messages-include="partials/templates/messages.tmpl.html"></div>
+                  </div>
+                </div>
+
+            </div>
+          </div>
+        </div>
+
+        <div class="col-md-6">
+          <div class="alert alert-info">
+            <h3>{{ 'VOUCHERS.COMPLEX.TITLE' | translate }}</h3>
+            <p>
+              {{ 'VOUCHERS.COMPLEX.DESCRIPTION' | translate }}
+            </p>
+          </div>
+        </div>
+      </div>
+
+      <!-- detail of transaction  -->
+      <div class="row">
+        <div class="col-md-12">
+          <div class="panel panel-default">
+            <div class="panel-heading">
+              <div class="btn pull-left">
+                <i class="glyphicon glyphicon-list-alt"></i>
+                {{ 'VOUCHERS.COMPLEX.TRANSACTION' | translate }}
+              </div>
+
+              <div class="pull-right">
+                <button type="button" class="btn btn-default" ng-click="ComplexVoucherCtrl.addVoucherItem()"
+                  data-button-add-item>
+                  {{ 'VOUCHERS.COMPLEX.ADD_ITEMS' | translate }}
+                  <i class="glyphicon glyphicon-plus"></i>
+                </button>
+              </div>
+
+              <div class="clearfix"></div>
+            </div>
+
+            <table class="table table-condensed table-striped">
+              <thead>
+                <tr>
+                  <th>{{ 'TABLE.COLUMNS.ACCOUNT' | translate }}</th>
+                  <th class="text-center">{{ 'TABLE.COLUMNS.DEBIT' | translate }}</th>
+                  <th class="text-center">{{ 'TABLE.COLUMNS.CREDIT' | translate }}</th>
+                  <th class="text-center">{{ 'VOUCHERS.COMPLEX.ENTITY' | translate }}</th>
+                  <th class="text-center">{{ 'TABLE.COLUMNS.REFERENCE' | translate }}</th>
+                  <th>&nbsp;</th>
+                </tr>
+              </thead>
+              <tbody>
+                <tr ng-repeat="row in ComplexVoucherCtrl.rows">
+                  <td style="width:30%">
+                    <input
+                      ng-model="row.account"
+                      ng-change="ComplexVoucherCtrl.checkRowValidity()"
```

Comment:
I wonder if you could make this function check the validity of a single row.  That way, you could write something like this:
```html
<input
  ng-change="ComplexVoucherCtrl.checkRowValidity($index)
>
```
which would check if the row at `$index` was valid.

At the moment, anytime a single input changes, you check that all properties in all rows are valid.  It might better to check that only the single row is valid, and set `row.valid = true` or `false` as needed.  Then, a global checker can just check that all rows in `ComplexVoucherCtrl.rows` have `row.valid` set to `true` and display a global error message or not.

This would allow you to set an icon (or background or border) on each row based on it's validity as well.

My proposal does not have to be implemented in this PR, but I am curious to hear what you think.  What do you think?